packages/core/src/loaders/imageLoader.ts

```diff
@@ -183,6 +196,126 @@ export function loadAndCacheImages(
   return allPromises;
 }
 
+/**
+ * Loads an image given an imageId and optional priority and returns a promise
+ * which will resolve to the loaded image object or fail if an error occurred.
+ * The image is stored in the cache.
+ *
+ * @param imageId -  A Cornerstone Image Object's imageId
+ * @param options - Options to be passed to the Image Loader
+ *
+ * @returns Image Loader Object
+ */
+export function createAndCacheDerivedImage(
+  imageId: string,
+  options: DerivedImageOptions
+): Promise<IImage> {
+  if (imageId === undefined) {
+    throw new Error(
+      'createAndCacheDerivedImage: parameter imageId must not be undefined'
+    );
+  }
+  const imagePlaneModule = metaData.get('imagePlaneModule', imageId);
```

Comment:
This should use the options.imageId first in case the imagePlaneModule has already been created, and then fallback to the base imageId module otherwise.

Reply:
I mean it is derived from that so I'm not sure what is the benefit there, the options.imageId is basically just for naming

Reply:
add a random generator

Reply:
createLocalImage

Reply:
refactor this outside


  ['imagePixelModule', 'imagePlaneModule', 'generalSeriesModule'].forEach(
    (type) => {
      derivedImageMetadataProvider.add(image.imageId, {
        type,
        metadata: metaData.get(type, referencedImageId),
      });
    }
  );

  derivedImageMetadataProvider.add(image.imageId, {
    type: 'referencedImageId',
    metadata: {
      referencedImageId,
    },
  });